bash-preexec.sh

```diff
@@ -244,32 +258,44 @@ __bp_preexec_invoke_exec() {
         return
     fi
 
-    # Invoke every function defined in our function array.
+    local preexec_ret_value
```

Comment:
The variable is set by `bash_preexec_invoke_preexec_functions`.

Now I'm afraid that you might not be familiar with the Bash's variable scoping. Bash's variable scope is dynamic scoping, so the callee sees all the local variables in the caller chain. You can see the following examples.

```bash
$ f2() { var=1; }
$ f1() { local var=0; f2; echo "$var"; }
$ f1
```

Using this dynamic scoping, we can pass/receive data through local variables between callers and callees without affecting the global context in Bash.

OK. Now I understood why you have asked about the underscore naming. Because of the dynamic scoping, the local variables defined in the function affect the context of all the callees, so we need to care about the name conflicts even if it is a local variable.

Reply:
a2216fc I have changed the strategy as suggested in the thread https://github.com/rcaloras/bash-preexec/pull/129#discussion_r858662337.